source/N4509.rst

```diff
@@ -0,0 +1,68 @@
+=======================================
+N4509 `constexpr is_always_lock_free()`
+=======================================
+
+:Author: Olivier Giroux
+:Contact: ogiroux@nvidia.com
+:Author: JF Bastien
+:Contact: jfb@google.com
+:Date: 2015-05-05
+:URL: https://github.com/jfbastien/papers/blob/master/source/4509.rst
+:Source: https://github.com/jfbastien/papers/blob/master/source/4509.cc
+
+The current design for ``std::atomic<T>`` affords implementations the critical
+freedom to revert to critical sections when hardware support for atomic
+operations does not meet the size or semantic requirements for the associated
+type ``T``. This preserves C++ support on aging hardware, when developers don't
```

Comment:
The 'when' is awkward.  Another form would be better.  Maybe "supports developers that don't".